lib/common/tests/iso8601/pcmk__time_format_hr_test.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <stdio.h>  // NULL
+
+#include <crm/common/iso8601_internal.h>
+#include <crm/common/unittest_internal.h>
+
+#define TEST_TIME pcmk__time_hr_new("2024-01-02 03:04:05 +00:00")
+
+/*!
+ * \internal
+ * \brief Assert that pcmk__time_format_hr() produced expected result
+ *
+ * \param[in] format     Time format string
+ * \param[in] expected   Assertion succeeds if result matches this
+ * \param[in] alternate  If this is not NULL, assertion may also succeed if
+ *                       result matches this
+ *
+ * \note This allows two possible results because different strftime()
+ *       implementations handle certain format syntax differently.
+ */
+static void
+assert_hr_format(const char *format, const char *expected,
+                 const char *alternate)
+{
+    pcmk__time_hr_t *hr = TEST_TIME;
+    char *result = pcmk__time_format_hr(format, hr);
+
+    pcmk__time_hr_free(hr);
+
+    if (expected == NULL) {
+        assert_null(result);
+        return;
+    }
+
+    assert_non_null(result);
+
+    if (alternate == NULL) {
+        assert_string_equal(result, expected);
+    } else {
+        assert_true((strcmp(result, expected) == 0)
+                    || (strcmp(result, alternate) == 0));
+    }
+
+    free(result);
+}
+
+static void
+null_format(void **state)
+{
+    assert_null(pcmk__time_format_hr(NULL, NULL));
+    assert_hr_format(NULL, NULL, NULL); // for pcmk__time_format_hr(NULL, hr)
+}
+
+static void
+no_specifiers(void **state)
+{
+    assert_hr_format("no specifiers", "no specifiers", NULL);
+    assert_hr_format("this has a literal % in it",
+                     "this has a literal % in it",
+                     // *BSD strftime() strips single %
```

Comment:
Apparently it also strips `%0`
```
14:24:32  FAIL: pcmk__time_format_hr_test 2 - no_specifiers
14:24:32  # "this has a literal 1 in it" != "this has a literal %01 in it"
```

Reply:
Yeah I figured another run would find something else